### ISSUE TITLE
profile-icons with web sources are not shown on the taskbar

### ISSUE THREAD

**Issue by torgeros** [microsoft/terminal#10552]
### Windows Terminal version (or Windows build number)

1.8.1521.0

### Steps to reproduce

**I am using WINDOWS 11, I do not know how this behaves in Win10.**

1. Create a new profile (or use an existing one)
2. For the icon, paste a weblink to an ico file (eg https://github.com/favicon.ico)
3. As you will see, that works fine (the icon gets shown in new tabs and in the settings menu)
![image](https://user-images.githubusercontent.com/67151313/124358125-29e67c80-dc1f-11eb-8610-91d3c0e3b58f.png)
4. keep winterm open
5. right-click onto the program in the taskbar
6. As you will see, the icons are not rendered here
![image](https://user-images.githubusercontent.com/67151313/124358169-57332a80-dc1f-11eb-89b5-7bee6bcfac0e.png)

### Expected Behavior

Profile Icons are rendered the same way everywhere across the device

### Actual Behavior

Profile Icons from the web are not rendered at all in the taskbar dialog

**Comment by zadjii-msft** [comment]
You know, I am confident that we have a dupe for this floating around on the repo, but I can't find it. The same thing happens when using an emoji as the icon for a profile. 

I suppose theoretically, we could actually download the image locally and set the path for the jumplist entry to that file.

**Comment by torgeros** [comment]
> I suppose theoretically, we could actually download the image locally and set the path for the jumplist entry to that file.

I someone is going to implement that, maybe also make copies of local images to an app-internal path. I found it somewhat annoying that I have to make sure that my custom local icons do not get deleted by accident.

Like for example when I download an image, set it as profile icon, and then clean up my downloads folder, the profile icon is gone.

**Comment by DHowett** [comment]
Yeah -- we should cache local *and* remote paths, honestly. We need to be extra careful to not cache old copies of local icons that have changed, but that should present little issue.

Technically -- and unrelately, mind you -- we should also be caching them in memory. Reloading every image every time the settings changes is a right pain. :smile:

**Comment by yash112-lang** [comment]
Hello @torgeros, is this issue is still open, I like to work on this issue. I am new to open source and want to contribute.

**Comment by torgeros** [comment]
I am not planning on working on this myself. No one else has posted something about them working on it.

Feel free to fork the repo and then open a pull request referencing this issue.

**Comment by JavierCCC** [comment]
This is still an issue

**Comment by N0xFF** [comment]
> This is still an issue

@JavierCCC try to re-pin it or reinstall.

**Comment by tejasness** [comment]
facing same issue

**Comment by tejasness** [comment]
> > This is still an issue
> 
> @JavierCCC try to re-pin it or reinstall.

didnt work

**Comment by yarkonev** [comment]
Facing this issue too
![Image](https://github.com/user-attachments/assets/aa2a2589-653b-4b5e-b51e-e1abc718cf3b)

**Comment by DHowett** [comment]
Soon, profile icons with web sources will not be shown at all. It's a trivial security risk.

**Comment by atiliosantosuu** [comment]
> Soon, profile icons with web sources will not be shown at all. It's a trivial security risk.

is that related something with wsl/install import/export? Cuz, it cant figure out the correct directory of the shorcut.ico, since 2.4.x manually i have to point the path of the profile icon for the distro who came from WSL install

**Comment by dr-jagulator** [comment]
It's the same thing.

![Image](https://github.com/user-attachments/assets/6ced2bd8-ac32-4076-953c-82965bc8d60e)

**Comment by soustruh** [comment]
I am sorry if this comment is already pointless, but the weird thing is:
- I see the icon missing, so I go to the Ubuntu profile settings.
- I see there is an icon "change" and an arrow to revert that change: `⮌`.
- I click the arrow, the URL immediately shows up in the appropriate text field and also it renders in the left sidebar and in the Icon row.
- I click Save and the icon disappears again. 👀

So is this a deliberate behavior in fact and is it caused by sanitizing the configuration JSON file from security risks as stated by @DHowett above?

**Comment by dr-jagulator** [comment]
@soustruh 
> * I see the icon missing, so I go to the Ubuntu profile settings.
> ...

Save the image by URL to a local folder (for example, Images), go to your Ubuntu profile and specify the location of the icon in the local folder.

**Comment by soustruh** [comment]
Yep, thanks. I already figured out that would work as a workaround and it really does, I just wanted to share my experience from saving the configuration (but then I read the comment above and realized everyone in this thread already knows this, but still made the comment just in case they don't) 😇
